starter_code/src/movies.js

```diff
@@ -1,15 +1,259 @@
 /* eslint no-restricted-globals: 'off' */
 
+
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
 
+const orderByYear = function orderByYear(movieArr) {
+
+    return movieArr.map(movie => movie).sort((a, b) => {
+
+        if (a.year === b.year) {
+
+            if (a.title < b.title) {
+
+                return -1
+
+            } else if (a.title > b.title) {
+
+                return 1
+
+            } else {
+
+                return 0;
+
+            }
+
+        }
+
+        return a.year - b.year;
+
+    })
+
+}
+
+console.log(`Iteration 1 | The order by year is: ${JSON.stringify(orderByYear(movies))}`);
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
 
+function howManyMovies(movArr) {
+
+    return movArr.filter(movie => movie.director === 'Steven Spielberg' && movie.genre.includes('Drama')).length;
+
+
+}
+
+
+console.log(`Iteration 2 | Steven Spielberg made ${howManyMovies(movies)} drama movies`);
+
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
 
+function orderTitlesAlphabetically(a, b) {
+
+    if (a.title < b.title) {
+
+        return -1
+
+    } else if (a.title > b.title) {
+
+        return 1
+
+    } else {
+
+        return 0;
+
+    }
+
+
+}
+
+const orderAlphabetically = function orderAlphabetically(movieArr) {
+
+    return movieArr.map(movie => movie).sort(orderTitlesAlphabetically)
+        .filter((element, index) => index < 20 ? element.title : null)
+        .map(element => element.title)
+
+}
+
+console.log(`Iteration 3 | These are the first 20 movies ordered alphabetically: ${orderAlphabetically(movies)}`);
+
 // Iteration 4: All rates average - Get the average of all rates with 2 decimals
 
+
+function calculateAverageRating(sumAverage, currentvalue, index, arr) {
+
+    if (currentvalue.rate === undefined || currentvalue.rate === null) {
+
+        return sumAverage;
+
+    }
+
+    return sumAverage + currentvalue.rate / arr.length;
+
+}
+
+const ratesAverage = function ratesAverage(movieArr) {
+
+
+    let averageNumber;
```

Comment:
You can also write it in one line:  let averageNumber = movieArr.reduce(calculateAverageRating, 0)

Reply:
oke, thanks